### PR TITLE
Use `class` instead of `lang` attribute and fix minified build

### PR DESCRIPTION
**Pull request by tomayac**
- Use `class` instead of `lang` attribute
- Fix the minified build
- Update dependencies

### CODE REVIEW

File: smd_min_entry.js [hunk 1]
```diff
@@ -28,9 +28,12 @@ export {
 	// Attr enum
 	HREF,
 	SRC,
-	LANG,
+	CLASS,
 	CHECKED,
 	START,
+	// Renderer
+	default_renderer,
+	logger_renderer,
```
Comment (thetarnav) [inline smd_min_entry.js:36]: The whole point of the min entry is to not include the example renderers.
They exist for starting point during development by "having something on the screen" before you go and make your own.
Normal entry you use with a bundler that will tree-shake them.
The min entry is usable from a cdn and should only have whats necessary.

**Review by thetarnav — COMMENTED** [review]
Why `class` instead of `lang`?
I've picked `lang` because it is unique and so can be easily associated with the code block.
`class` is ambiguous.
If you need to use a different attribute, you can change that in your renderer.

**Comment by tomayac** [comment]
> Why `class` instead of `lang`? I've picked `lang` because it is unique and so can be easily associated with the code block. `class` is ambiguous. If you need to use a different attribute, you can change that in your renderer.

If `lang` is your default, you possibly block `lang` forever if the HTML standard ever wanted to define it. Something similar has [happened before with Array flattening](https://developer.chrome.com/blog/smooshgate).

File: smd_min_entry.js (hunk 1 above)
Comment (tomayac) [reply smd_min_entry.js:36]: Alright, not exposing the `logger_renderer` then.

**Comment by tomayac** [comment]
> > Why `class` instead of `lang`? I've picked `lang` because it is unique and so can be easily associated with the code block. `class` is ambiguous. If you need to use a different attribute, you can change that in your renderer.
> 
> If `lang` is your default, you possibly block `lang` forever if the HTML standard ever wanted to define it. Something similar has [happened before with Array flattening](https://developer.chrome.com/blog/smooshgate).

The `class` is also what popular syntax highlights [like Prism use](https://prismjs.com/#basic-usage:~:text=%3Cpre%3E%3Ccode%20class%3D%22language%2Dcss%22%3Ep%20%7B%20color%3A%20red%20%7D%3C/code%3E%3C/pre%3E).

**Comment by tomayac** [comment]
Actually, I'm stupid in what I said in https://github.com/thetarnav/streaming-markdown/pull/6#issuecomment-2405598025. The [`lang` attribute is already defined](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang), and its value must be in the format defined in [RFC 5646](https://datatracker.ietf.org/doc/html/rfc5646). So using something like `lang="javascript"` is actually wrong. The move to `class` makes even more sense given this background.

**Comment by tomayac** [comment]
Friendly ping on this PR. Is this something you're considering? Thanks!

**Comment by thetarnav** [comment]
Right.. yes.
The artice you've sent was terrifying and eye-oppening.
I doubt this lib will block anything but I'm also not one to decide what `lang` attr should mean.
So while the dom renderer should just set it as `class` or `data-lang`, I don't want to rename the `LANG` and `Attr` enums. They should stay as they are. `attr_to_html_attr` is where the translation happens. So far I tried to align the enum with html, but that doesn't have to always be so.
Also I'm still on the fence about exporting `default_renderer` in the min entry. If you are gonna use a cdn, copy the code for the dom renderer, it's just there as an example. Maybe it's a bit weird to provide a library that only does half the job, but in my mind you will end up making a custom renderer anyway, once you need to change anything. (like the attr name for code blocks :))

**Comment by tomayac** [comment]
Please see the current version of the PR, moved `CLASS` back to `LANG`, and only doing the translation to `class` in the `attr_to_html_attr()` function.

I think exporting the `default_renderer` makes a lot of sense. It certainly worked well enough for [my use case](https://glitch.com/edit/#!/ai-streaming-parser?path=script.js%3A9%3A45) (note that this is dependent on a [behind-a-flag API](https://developer.chrome.com/docs/ai) in Chrome).

**Pull request merged**